ConnectorArchicad/AddOn/Sources/AddOn/Commands/GetModelForElements.cpp

```diff
@@ -79,7 +79,9 @@ static void GetModelInfoForElement (const Modeler::Elem& elem,
 		// edges
 		for (ULong edgeIdx = 0; edgeIdx < body.GetEdgeCount (); ++edgeIdx) {
 			const EDGE& edge = body.GetConstEdge (edgeIdx);
-			modelInfo.AddEdge (ModelInfo::EdgeId (edge.vert1 + vetrexOffset, edge.vert2 + vetrexOffset), ModelInfo::EdgeData (ModelInfo::VisibleEdge, edge.pgon1, edge.pgon2));
+			// send only edges which have no polygons (only 3D lines)
+			if (edge.pgon1 == Brep::MeshBrep::Edge::InvalidPgonIdx && edge.pgon2 == Brep::MeshBrep::Edge::InvalidPgonIdx)
+				modelInfo.AddEdge (ModelInfo::EdgeId (edge.vert1 + vetrexOffset, edge.vert2 + vetrexOffset), ModelInfo::EdgeData (ModelInfo::VisibleEdge, edge.pgon1, edge.pgon2));
```

Comment:
Not sure I fully understand the reasoning behind this, or what it is actually doing.

Could you elaborate more on what other "non 3D lines" were getting sent? And why are they not necessary?

Reply:
The are vertices, edges and polygons for meshes in Archicad. The edge.pgon1 and edge.pgon2 are the indices of the neighbour polygons of the edge (there are max two of them). If the edge.pgon1 and edge.pgon2 are InvalidPgonIdx, it means that edge is just a line in the 3D model with not polygons attached. This is the case for grid elements. So only these kind of edge/lines are sent to speckle. (Edge status for Mesh elements are not supported yet.)

Reply:
Oohhh, ok I get it now. This sounds legit 🙌🏼 flagging this as resolved